media/windows/wintasksmon/wintasksmon.py

```diff
@@ -0,0 +1,439 @@
+                                if not run_task_status:
+                                    slk_message = slk_message + \
+                                                  f'\nError occurred attempting to restart task:  {task_to_check["task"]}'
+                        else:
+                            print(f'*** Task completed successfully on last run at:  {task.LastRunTime}')
+                            logger.info(f'*** Task completed successfully on last run at:  {task.LastRunTime}')
+                            # slk_message = f'Task [{task_to_check["task"]}] completed successfully on last run at:' \
+                            #               f'  {task.LastRunTime} '
+                            tasks_last_run_state.append({"task": task_to_check["task"],
+                                                         "last_state": "success"})
+                    elif task_to_check["status"] == 'running':
+                        if task.State != 4:
+                            print(f'Task {task_to_check["task"]} is not running!!!')
+                            logger.info(f'Task {task_to_check["task"]} is not running!!!')
+                            slk_message = f'Task [{task_to_check["task"]}] is not running!!!'
+                            if task_to_check["run_flag"] == 'restart':
+                                run_task_status = run_task(task)
+                                if not run_task_status:
+                                    slk_message = slk_message + \
+                                                  f'\nError occurred attempting to restart task:  {task_to_check["task"]}'
+                            tasks_last_run_state.append({"task": task_to_check["task"],
+                                                         "last_state": "not running"})
+                        else:
+                            print(f'Task {task_to_check["task"]} is running...')
+                            # print(f'...sending Slack message...')
+                            logger.info(f'Task {task_to_check["task"]} is running...')
+                            # logger.info(f'...sending Slack message...')
+                            # slk_message = f'Task [{task_to_check["task"]}] is running...'
+                            tasks_last_run_state.append({"task": task_to_check["task"],
+                                                         "last_state": "running"})
+                    else:
+                        print(f'Specified task status to check is invalid!')
+                        logger.error(f'Specified task status to check is invalid!')
+                        slk_message = f'Invalid status to check specified for task [{task_to_check["task"]}]...'
+
+                    if slk_message != "":
+                        # Check the last run-status; if this matches the current state, don't send a Slack alert
+                        send_slack = check_last_run_state(json_last_run_state_list, task_to_check, task)
```

Comment:
I see that this is checking against `json_last_run_state_list`, which was loaded once, at the beginning of the run.  Does this value need to be repeatedly loaded after each iteration?

Reply:
If I understand your question, the answer is 'no'...
The last-run-state is loaded (from the last execution) once at the beginning of the run.  This call is simply checking the current task's run state against the last iteration to see if it has changed (more correctly, hasn't changed...thereby suppressing the notification).

Reply:
Got it!